components/brave_new_tab_ui/data/backgrounds.ts

```diff
@@ -14,7 +14,7 @@ export const images: NewTab.BraveBackground[] = [{
 // then you must also update `script/generate_licenses.py`
 
 export const updateImages = (newImages: NewTab.BraveBackground[]) => {
-  if (!newImages) {
+  if (!newImages?.length) {
```

Comment:
Obviously, I made a mistake here. An empty [] is evaluated true.

Reply:
why don't use `!newImages.length` here if empty array is passed?

Reply:
Then the default image will be gone.

Reply:
Yeah, it's kinda confusing. Array.splice() will remove from the 0 to image.length count, and then insert given items to the array. So, if newImages are empty, it'll only remove all existing items and not insert new items.

Reply:
Aha, I was mistaken about your comment. Yeah, we can remove "?". newImages can't be undefined or null so we don't need it.

Reply:
Removed! Thanks :)